momento-sdk/src/main/java/momento/sdk/responses/topic/SubscriptionState.java

```diff
@@ -0,0 +1,39 @@
+package momento.sdk.responses.topic;
+
+public class SubscriptionState {
+  private Runnable unsubscribeFn;
+  private Integer lastTopicSequenceNumber;
+  private boolean isSubscribed;
+
+  public SubscriptionState() {
+    this.unsubscribeFn = () -> {};
+    this.isSubscribed = false;
+  }
+
+  public int getResumeAtTopicSequenceNumber() {
+    return (lastTopicSequenceNumber != null ? lastTopicSequenceNumber : -1) + 1;
+  }
+
+  public void setSubscribed() {
+    this.isSubscribed = true;
+  }
+
+  public void setUnsubscribed() {
```

Comment:
When would I use this vs `unsubscribe()`? You might want to add some javadocs here since there are similarly named functions that do different things.

Reply:
setUnsubscribed sets the subscribed object to false when unsubscribe function is called. Added java docs which should make it more clear.